packages/eas-cli/src/build/context.ts

```diff
@@ -122,3 +136,43 @@ export function createBuildContext<T extends Platform>({
     buildProfile,
   };
 }
+
+function getDevClientEventProperties(
+  platform: Platform,
+  commandCtx: CommandContext,
+  buildProfile: AndroidBuildProfile | IosBuildProfile
+): { dev_client: boolean; dev_client_version?: string } {
+  const devClientVersion = tryGetDevClientVersion(commandCtx.projectDir);
+
+  if (buildProfile.workflow === Workflow.MANAGED) {
+    return {
+      dev_client:
+        buildProfile.buildType === Android.ManagedBuildType.DEVELOPMENT_CLIENT ||
+        buildProfile.buildType === Ios.ManagedBuildType.DEVELOPMENT_CLIENT,
+      dev_client_version: devClientVersion,
+    };
+  } else if (platform === Platform.ANDROID) {
+    const { gradleCommand } = buildProfile as AndroidGenericBuildProfile;
+    return {
+      dev_client: Boolean(devClientVersion && gradleCommand?.includes('Debug')),
+      dev_client_version: devClientVersion,
+    };
+  } else if (platform === Platform.IOS) {
+    const { schemeBuildConfiguration } = buildProfile as IosGenericBuildProfile;
+    return {
+      dev_client: Boolean(devClientVersion && schemeBuildConfiguration === 'Debug'),
```

Comment:
technically it's possible to have Debug as the default buildConfiguration in xcode, so it will build dev client if schemeBuildConfiguration is not defined

but it be would quite a weird thing to do so I don't think we need to handle that case just for analytics